### DISCUSSION TITLE
Accessing getUserInfo() with the wagmi-connector plugin

### DISCUSSION THREAD

**Discussion by sultanpeyek** [Web3Auth/Web3Auth#1025] [Help]
Hello everyone and @yashovardhan,

I have a question about the interaction between wagmi-connector and web3auth. I was able to connect a wallet to web3auth using the [wagmi-connector](
https://github.com/Web3Auth/web3auth-wagmi-connector) plugin, but I'm not sure how to call the `await web3Auth?.getUserInfo()` command. Could anyone provide some guidance on this?

Thank you very much.

**Comment by yashovardhan** [comment]
Hey @sultanpeyek 

Thanks for the question. Sorry for not getting back to you earlier on discord. I looked around the SDK, as I see currently that function is not exposed because of wagmi limitations. 

@himanshuchawla009 is it possible for us to expose that function in the connector?

**Reply by sultanpeyek** [reply]
Hey @yashovardhan and @himanshuchawla009 I'm fine, it would be nice if it's possible. Wagmi is a powerful tool, and I think many people will use it in the future. I don't want to lose the functionality of the web3auth provider functions at the same time.

If I may add, I also want the ability to export private keys. I'm fine with using the opposite approach, where I use the default web3auth provider to connect the wallet and then inject the connector/signer into wagmi. All I really need is the benefit of using the wagmi hooks.

**Comment by krisgrm** [comment]
Hey I am also looking for this option

**Comment by thedonmon** [comment]
I’m also looking for this functionality as well. The only way I could get this to work is by reintroducing the web3authcontext like the example docs and then passing the web3auth instance created for WAGMI in the context provider. You basically can only get user info once tho after successful signup but I believe it also interferes with some other functionality.

**Reply by webthethird** [reply]
I know this is a bit old, but could you elaborate on this solution, and possibly share some code?

**Reply by thedonmon** [reply]
Ill have to dig it up, decided to ditch web3auth due to this limitation/hope this issue is fixed by now... I should still have the branch

**Reply by webthethird** [reply]
Don't worry about it, I actually solved it myself

**Comment by devdomsos** [comment]
Wait that means I cannot get userInfo if I use wagmi-connector implementation? :)

**Comment by lybenson** [comment]
try this
```ts
import { useAccount } from 'wagmi'

const account = useAccount()
const connector = account.connector
const web3Auth = connector.options.web3AuthInstance

await web3Auth.getUserInfo()
```

**Reply by codescrafter** [reply]
doesnt work
options doesnt exits on connector
